tests/Feature/ProductTest.php

```diff
@@ -77,9 +77,10 @@
         'status' => false,
     ] );
 
-        $product = Product::where('name', 'PDU Update')->first();
-        $this->assertTrue(
-              $product->categories->whereIn('id', $categoryIds)->isNotEmpty());
+    $firstCategoryId = reset($categoryIds); 
+    $this->assertDatabaseHas('category_product', [
+        'category_id' => $firstCategoryId,
+    ]);
```

Comment:
$categories = Category::factory(3)->create();

this give you the collection so play with this. passing only category Ids & same check first for database has id.